webapp/graphite/worker_pool/pool.py

```diff
@@ -0,0 +1,151 @@
+import time
+from Queue import Queue, Full, Empty
+from threading import Thread, Lock
+
+from django.conf import settings
+
+from graphite.logger import log
+
+# we'll never want more than one instance of Pool
+init_lock = Lock()
+is_initialized = False
+
+
+def get_pool():
+  with init_lock:
+    instance = getattr(get_pool, 'instance', None)
+    if instance is None:
+      instance = Pool()
+      setattr(get_pool, 'instance', instance)
+      is_initialized = True
+  return instance
+
+
+def stop_pool():
+  with init_lock:
+    if not is_initialized:
+      return
+
+    get_pool().shutdown()
+
+
+class Pool(object):
```

Comment:
The API is basically multiprocessing.Pool. It's not documented because they want people to use concurrent.futures.ThreadPoolExecutor (Python 3 only). It's exposed so it's supposed to be stable